R/repr_matrix_df.r

```diff
@@ -161,7 +173,10 @@ repr_latex.matrix <- function(obj, ..., colspec = getOption('repr.matrix.latex.c
 	obj <- latex.escape.names(obj)
 	
 	# escape LaTeX special characters if necessary
-	if (any(apply(obj, 2L, any.latex.specials))) {
+	need_to_excape_col <- vapply(seq_len(ncol(obj)), function(col_index) {
+			any.latex.specials(obj[, col_index])
+		}, FUN.VALUE = TRUE)
```

Comment:
It's a little strange, but in issue #30, using `apply` calls `as.matrix.data.frame`, which errors out because of the weird name issues I described in that issue thread.  In contrast, using `seq_len` to index over columns doesn't convert the data.frame to a matrix, so avoids the error.  I'm not sure how you want to handle this; it might be an issue in `dplyr` or just an issue with my understanding of how `dplyr` handles names.